processors/base/writers/document.py

```diff
@@ -70,14 +58,69 @@ def write_document(conn, document):
         )
         return None
 
-    # Write object
-    conn['database']['documents'].upsert(obj, ['id'], ensure=False)
+    # Write object and relationships
+    try:
+        db.begin()
+
+        db['documents'].upsert(obj, ['id'], ensure=False)
+
+        trial_id = document.get('trial_id')
+        if trial_id:
+            trials_documents = {
+                'trial_id': trial_id,
+                'document_id': obj['id'],
+            }
+            db['trials_documents'].insert_ignore(
```

Comment:
`insert_ignore` inserts a new row if one with the keys doesn't exist (here `['trial_id', 'document_id']`), or ignores if one exist (i.e. doesn't update or do anything). In contrast with the more common `upsert`, which updates an existing row if it exists, `insert_ignore` does nothing in that case.

This is because `trials_documents` only has columns `trial_id` and `document_id`, so updating it doesn't make sense (it would update it with the same values it has now, not changing anything)